rfcs/039-table-data-quality-checks.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: table_data_quality_checks
+- Start Date: 2021-07-16
+- RFC PR: [amundsen-io/rfcs#XX](https://github.com/amundsen-io/rfcs/pull/XX)
+# Table Data Quality Checks
+
+## Summary
+This RFC will define a generic interface for accessing data quality checks in Amundsen. If available, we aim to expose
+the status of these data quality checks in the Amundsen Table Details page.
+
+
+## Terms
+- Data Quality Check - Like unit tests for code, data quality checks are assertions that try to guarantee certain 
+  characteristics of data. For example a data quality check could be as simple as asserting that a column is not null,
+  or implement some complex custom logic.
+
+
+## Motivation
+While Amundsen is not currently building unit tests for data, we can empower our users by allowing integrations 
+with a generic external data quality service. Exposing the status of data quality checks in Amundsen can give users 
+an at-a-glance summary of the table's quality. This can help a user to quickly decide if they want to use this table.
+
+## Product Details
+Users will see a new section on the left panel of the Table Details page titled `Data Quality Checks`. There will be
+a message and icon for the current status of these checks, if they exist. A passing table might say 
+`✅ 10 of 10 checks passed` and a failing table might say `❌ 3 of 10 checks failed`. Additionally, the user may see 
+the last run timestamp, and a link to the external data quality service, if available.
+
+## UX Explanation
+A new section will be added to the left side of the Table Details Page.
+
+![Data Quality Preview](../assets/039/data_quality.png)
+
+## Technical Details
+Table Data Quality Checks will be implemented as an optional feature on Amundsen Frontend. This feature will be hidden 
+behind a config flag and will use a generic client much like the existing data preview client. The generic client should
+implement the following interface: 
```

Comment:
Any reason why we are not considering extending Table model with quality check entity?

I think there are a lot of tools that can be used to run DQ checks but don't provide integration in the form of API to query those results and this approach limits us to only tools that would have it.

Reply:
I am not against extending the Table model and defining databuilder tasks to ingest DQ checks from various providers. I think that's a great idea with many merits.

However this RFC isn't intended to define a fully-fledged solution, but rather proposes a simple integration via an API interface. 

That being said I have [this old PR](https://github.com/amundsen-io/amundsen/pull/1073) which was intended to be merged into a private repo, until I was asked to contribute it to OSS. So perhaps this solution is a little narrow minded to retrofit an existing PR.